### PR TITLE
Add pylance

### PR DESCRIPTION
**Pull request by yhori991**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by yhori991** [comment]
I'm happy to be listed as the maintainer of this recipe 👍

**Comment by yhori991** [comment]
@conda-forge/staged-recipes this PR is ready for review. 
I could not make a practical `win-64` build but still, it is hopefully helpful for those who need this package on conda ecosystem. lancedb/lance#636

File: recipes/pylance/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{% set name = "pylance" %}
+{% set version = "0.9.8" %}
+{% set build_number = environ.get('GIT_DESCRIBE_NUMBER', '0') %}
```
Comment (xhochy) [inline recipes/pylance/meta.yaml:3]:
```suggestion
```

File: recipes/pylance/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+source:
+  url: https://github.com/lancedb/lance/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: b1152fd112214e2974280964ba934202e7e8b736cd768177d0ff25dc80663392
+
+build:
+  number: 0
+  # There's a protocol buffer related error...
+  # This recipe skips windows until it gets fixed...!
+  # https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=862176&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=7c0f8eae-6d6f-51bf-636f-73a1a7fb1bc4
+  skip: true  # [py<38 or win]
```
Comment (xhochy) [inline recipes/pylance/meta.yaml:18]: Please remove the skip and let's have a look together at this.

File: recipes/pylance/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  build:
+    - python                              # [build_platform != target_platform]
+    # there is no cross-python for linux-64 -> win-64
+    - cross-python_{{ target_platform }}  # [build_platform != target_platform and not target_platform == "win-64"]
+    - crossenv                            # [build_platform != target_platform]
+    - maturin >=1.4.0,<2                  # [build_platform != target_platform]
+    - {{ compiler('c') }}
+    - {{ compiler('rust') }}
+    - cargo-bundle-licenses
+    - protobuf
```
[the remark this answers is not in the document]
Comment (yhori991) [reply recipes/pylance/meta.yaml:30]: libprotobuf must be okey for providing `protoc`.

File: recipes/pylance/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  host:
+    # this is a hacky way to do cross-compilation from linux to windows
```
[the remark this answers is not in the document]
Comment (yhori991) [reply recipes/pylance/meta.yaml:31]: This is a setting that I added as a last resort to build a binary for Windows by cross-compiling on Linux. We can remove this to make it simpler.

File: recipes/pylance/meta.yaml (hunk 2 above)
Comment (yhori991) [reply recipes/pylance/meta.yaml]: I've updated the file to remove the cross-compile related settings and the skip part for `win`.

**Comment by yhori991** [comment]
Thanks for the comments and reviews @xhochy. 

Aside from `win-64` issues, the current change looks much cleaner and better.
The error log is in the L1538 of this log down below. 

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=862707&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=7c0f8eae-6d6f-51bf-636f-73a1a7fb1bc4&l=1538

It is a `protoc` error that can be seen neither in OSX nor Linux host environment,  which is saying `format.proto` was not found. This error message is mysterious to my eyes since the original repo does not have `format.proto` from the beginning 🤔

**Comment by yhori991** [comment]
I've just noticed several code issues and made two updates too. 

- fixed the project home URL
- strictly align maturin build parameters of UNIX host and Windows host

The win-64 build issue is still not resolved but it might be now easier to narrow down the cause.

**Comment by yhori991** [comment]
Luckily for us, bumping the package version up to 0.9.10 solved the win-64 build issue it seems.

File: recipes/pylance/conda_build_config.yaml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+macos_min_version:             # [osx]
+  - 10.12                      # [osx]
```
Comment (xhochy) [inline recipes/pylance/conda_build_config.yaml:2]: You shouldn't need this.
```suggestion
```

File: recipes/pylance/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  requires:
+    - pip
+
+about:
+  home: https://github.com/lancedb/lance
+  summary: python wrapper for Lance columnar format
+  license: Apache-2.0
+  license_file:
+    - LICENSE
+    - python/LICENSE
```
[the remark this answers is not in the document]
Comment (yhori991) [reply recipes/pylance/meta.yaml:46]: `cargo-bundle-licenses` output..! Thanks for mentioning this.

**Comment by yhori991** [comment]
Bumping the version number to v0.9.12 breaks the linux-64 build. 

From the CI log, there seems to be a overlinking issue on `xz` library.

> conda_build.exceptions.OverLinkingError: overlinking check failed 
> ["  ERROR (pylance,lib/python3.8/site-packages/lance/lance.abi3.so): .. but ['conda-forge/linux-64::xz==5.2.6=h166bdaf_0'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)"]

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=869517&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1455

**Comment by yhori991** [comment]
I'll first check how it goes on v0.9.11 and narrow down the cause of the issue.

**Comment by yhori991** [comment]
On the surface level, there are no significant changes on `Cargo.toml' files that seem to be relevant to the linking issue above. I'll think about adding `xz` as a runtime dependency to see if it works (there's still a chance that it might be truly missing dependency)

**Comment by yhori991** [comment]
It is building `xz2` in here, which is a binding to `liblzma` (xz utils). 

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=870007&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1292 

It might be legitimate to add xz as a dependancy.

File: recipes/pylance/meta.yaml [hunk 7]
```diff
@@ -28,6 +28,7 @@ requirements:
     - numpy >=1.22
     - pyarrow >=12
     - pyarrow-hotfix
+    - xz  # [linux-64]
```
Comment (xhochy) [inline recipes/pylance/meta.yaml:31]: This should be in `host`

File: recipes/pylance/meta.yaml [hunk 8]
```diff
@@ -28,6 +28,7 @@ requirements:
     - numpy >=1.22
     - pyarrow >=12
     - pyarrow-hotfix
```
Comment (xhochy) [inline recipes/pylance/meta.yaml:30]: We probably won't support installing `pyarrow-hotfix` with newer arrow versions. You should omit it here.
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Review by xhochy — APPROVED** [review]

**Pull request merged**